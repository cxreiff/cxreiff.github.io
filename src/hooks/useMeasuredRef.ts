import { useRef, useState, useEffect } from 'react'

export function useMeasuredRef<T extends HTMLElement>() {
    const ref = useRef<T | null>(null)
    const [rect, setRect] = useState<DOMRect>(new DOMRect())

    useEffect(() => {
        const resizeObserver = new ResizeObserver(entries => {
            entries.map(entry => {
                setRect(entry.target.getBoundingClientRect())
            })
        })

        if(ref.current) {
            resizeObserver.observe(ref.current)
        }
        return () => resizeObserver.disconnect()
    }, [ref.current])

    return [ref, rect] as const
}
