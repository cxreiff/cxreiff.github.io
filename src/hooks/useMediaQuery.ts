import { useEffect, useState } from "react";

export const MEDIA_QUERIES = {
  LARGER_THAN_XS: "(min-width: 385px)",
  LARGER_THAN_SM: "(min-width: 768px)",
  LARGER_THAN_MD: "(min-width: 1400px)",
  MOBILE: "only screen and (hover: none) and (max-width: 768px)",
} as const;

export const useMediaQuery = (
  query: (typeof MEDIA_QUERIES)[keyof typeof MEDIA_QUERIES]
) => {
  const [matches, setMatches] = useState(window.matchMedia(query).matches);

  useEffect(() => {
    const mediaQuery = window.matchMedia(query);
    const listener = () => setMatches(mediaQuery.matches);
    mediaQuery.addEventListener("change", listener);
    return () => mediaQuery.removeEventListener("change", listener);
  }, [matches]);

  return matches;
};
