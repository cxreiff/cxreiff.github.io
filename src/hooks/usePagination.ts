import { useState, Dispatch, SetStateAction } from "react";

export const usePagination = (): {
  page: number;
  totalPages: number;
  setTotalPages: Dispatch<SetStateAction<number>>;
  previousPage: (numberOfPages: number) => void;
  nextPage: (numberOfPages: number) => void;
  hasPreviousPage: (page: number) => boolean;
  hasNextPage: (page: number) => boolean;
  goToPage: (pageNumber: number) => void;
} => {
  const [page, setPage] = useState(0);
  const [totalPages, setTotalPages] = useState(1);

  const previousPage = (numberOfPages = 1) => {
    setPage(Math.max(page - numberOfPages, 0));
  };

  const nextPage = (numberOfPages = 1) => {
    setPage(Math.min(page + numberOfPages, totalPages - 1));
  };

  const hasPreviousPage = (pageNumber = page) => pageNumber > 0;

  const hasNextPage = (pageNumber = page) => pageNumber + 1 < totalPages;

  const goToPage = (pageNumber: number) => {
    if (hasPreviousPage(pageNumber + 1) && hasNextPage(pageNumber - 1)) {
      setPage(pageNumber);
    }
  };

  return {
    page,
    totalPages,
    setTotalPages,
    previousPage,
    nextPage,
    hasPreviousPage,
    hasNextPage,
    goToPage,
  };
};
