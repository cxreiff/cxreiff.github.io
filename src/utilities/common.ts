
export const isMobile = () => window.matchMedia(
    "only screen and (hover: none) and (max-width: 768px)"
).matches
