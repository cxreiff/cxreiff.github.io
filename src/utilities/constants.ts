export const OBJECT_STORE_ASSETS_URL = 'https://jaxreiff-static.sfo3.cdn.digitaloceanspaces.com/'