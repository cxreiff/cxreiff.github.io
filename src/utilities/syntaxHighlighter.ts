import { CSSProperties } from 'react'
import { PrismLight } from 'react-syntax-highlighter'
import { prism } from 'react-syntax-highlighter/dist/esm/styles/prism'

/**
* Import and register languages for highlighting as-needed.
* (Importing individually for the purpose of keeping bundle size low)
*/

import tsx from 'react-syntax-highlighter/dist/esm/languages/prism/tsx'
PrismLight.registerLanguage('tsx', tsx)

import scss from 'react-syntax-highlighter/dist/esm/languages/prism/scss'
PrismLight.registerLanguage('json', scss)

export const SyntaxHighlighter = PrismLight
export const syntaxStyle = prism as CSSProperties
