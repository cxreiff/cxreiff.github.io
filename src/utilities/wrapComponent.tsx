import React, { ComponentType, ReactElement } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import store from '~/src/app/store';
import history from '~/src/app/history';

const wrapComponent = (Component: ComponentType, props = {}): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Component {...props} />
      </ConnectedRouter>
    </Provider>
  )
}

export default wrapComponent
