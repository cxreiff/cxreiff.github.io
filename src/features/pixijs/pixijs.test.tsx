import { render } from '@testing-library/preact'

import * as useMeasuredRef from '~/src/hooks/useMeasuredRef'
import * as common from '~/src/utilities/common'

import Pixijs from './pixijs.component'

jest.spyOn(useMeasuredRef, 'useMeasuredRef').mockImplementation(() => [
    { current: null },
    {
        x: 0,
        y: 0,
        width: 0,
        height: 0,
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        toJSON: jest.fn(),
    },
])
jest.spyOn(common, 'isMobile').mockImplementation(() => true)

test('should render', () => {
    const { container } = render(<Pixijs />)

    expect(container).toMatchSnapshot()
})
