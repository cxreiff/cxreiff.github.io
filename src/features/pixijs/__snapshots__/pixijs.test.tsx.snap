// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="frame pixijs"
  >
    <canvas
      style="height: 0px;"
    />
    <svg
      viewBox="0 0 1.7777777777777777 1"
    />
  </div>
  <div
    class="controls"
  >
    <button>
      FIRE
    </button>
    <button>
      TURN_LEFT
    </button>
    <button>
      TURN_RIGHT
    </button>
    <button>
      FORWARD
    </button>
    <button>
      BACKWARD
    </button>
  </div>
</div>
`;
